extras/concept_mapping_coverage_per_domain.R

```diff
@@ -0,0 +1,98 @@
+library("dplyr")
+library("jsonlite")
+library("ggplot2")
+
+setwd('/my/path/to/dqd/')
+
+dqd_filename <- 'dqd.json'
+
+# Load data
+result <- jsonlite::fromJSON(dqd_filename)
+check_results <- result$CheckResults %>%
+  select(CHECK_NAME, CDM_TABLE_NAME, CDM_FIELD_NAME, 
+         NUM_VIOLATED_ROWS, NUM_DENOMINATOR_ROWS, PCT_VIOLATED_ROWS)
+
+# Add fields
+coverage_results <- check_results %>%
+  filter(CHECK_NAME %in% c("standardConceptRecordCompleteness", "sourceValueCompleteness")) %>%
+  # Not interested in era's as these are all derived
+  filter(!(CDM_TABLE_NAME %in% c("DRUG_ERA", "DOSE_ERA", "CONDITION_ERA"))) %>%
+  mutate(
+    # First check is over all records, second over the unique source terms
+    coverageType = recode(CHECK_NAME, 
+                          standardConceptRecordCompleteness = "Records",
+                          sourceValueCompleteness = "Terms"),
+    # Coverage is rows not failing
+    coveragePct = 1 - PCT_VIOLATED_ROWS,
+    # Naming of domains
+    domain = recode(CDM_TABLE_NAME, 
+                    VISIT_OCCURRENCE = "VISIT",
+                    CONDITION_OCCURRENCE = "CONDITION",
+                    PROCEDURE_OCCURRENCE = "PROCEDURE",
+                    DRUG_EXPOSURE = "DRUG",
+                    DEVICE_EXPOSURE = "DEVICE"
+    ),
```

Comment:
just in case, if you want to make it general you can use `separate`:

`# close mutate parenthesis`
`separate(CDM_TABLE_NAME, into="domain", sep="_", extra="drop", remove=FALSE) %>%`
`# open new mutate`
`mutate(`